lib/public_key/src/pubkey_ocsp.erl

```diff
@@ -92,16 +106,23 @@ match_single_response(_IssuerName, _IssuerKey, _SerialNum, []) ->
 match_single_response(IssuerName, IssuerKey, SerialNum,
                       [#'SingleResponse'{
                           certID = #'CertID'{hashAlgorithm = Algo} = CertID} =
-                           Response | Responses]) ->
+                           SingleResponse | Tail]) ->
+    #'SingleResponse'{thisUpdate = ThisUpdate,
+                      nextUpdate = NextUpdate} = SingleResponse,
     HashType = public_key:pkix_hash_type(Algo#'AlgorithmIdentifier'.algorithm),
     case (SerialNum == CertID#'CertID'.serialNumber) andalso
         (crypto:hash(HashType, IssuerName) == CertID#'CertID'.issuerNameHash) andalso
-        (crypto:hash(HashType, IssuerKey) == CertID#'CertID'.issuerKeyHash) of
+        (crypto:hash(HashType, IssuerKey) == CertID#'CertID'.issuerKeyHash) andalso
+        verify_past_timestamp(ThisUpdate) == ok andalso
+        verify_next_update(NextUpdate) == ok of
         true ->
-            {ok, Response};
+            {ok, SingleResponse};
         false ->
-            match_single_response(IssuerName, IssuerKey, SerialNum, Responses)
-    end.
+            match_single_response(IssuerName, IssuerKey, SerialNum, Tail)
+    end;
+match_single_response(IssuerName, IssuerKey, SerialNum,
+                      [_BadSingleResponse | Tail]) ->
```

Comment:
Is it worth doing some common verify_data function in pubkey_cert.

Reply:
I'm not sure what we could gain here ...
`public_key:verify` is used already in `do_verify_signature`...

Am I missing something? pls explain.

Reply:
This was a long time ago. I think I was think that the logic of verify_past_timstamp and verify_next_update also exists in pubkey_cert module.

Reply:
Well potentially, pubkey_cert could be refactored and make use of atomic time comparison functions introduced in pubkey_ocsp.

But the overlap is relatively small and time verifications for OCSP are a bit specific not matching 1:1 with pubkey_cert, I don't think it makes sense now. Will resolve that one without any change.